gguf-py/scripts/gguf_dump.py

```diff
@@ -254,16 +254,26 @@ def dump_markdown_metadata(reader: GGUFReader, args: argparse.Namespace) -> None
                 value = str(field.parts[-1][0])
         else:
             if field.types[0] == GGUFValueType.ARRAY:
+                displayed_values = 0
                 curr_type = field.types[1]
                 if curr_type == GGUFValueType.STRING:
                     render_element = min(5, total_elements)
                     for element_pos in range(render_element):
-                        value += repr(str(bytes(field.parts[-1 - element_pos]), encoding='utf-8')[:5]) + (", " if total_elements > 1 else "")
+                        truncate_length = 30
+                        value_string = repr(str(bytes(field.parts[-1 - (total_elements - element_pos - 1) * 2]), encoding='utf-8'))
+                        if len(value_string) > truncate_length:
+                            value += value_string[:truncate_length // 2] + "..." + value_string[-truncate_length // 2:]
```

Comment:
> The initial gguf dump didn't match the output of main.cpp so must have read it wrong. Adjusted the python script until it matched

I'm not sure the dump should truncate the values. But since this is the markdown output, maybe it's acceptable.

Reply:
Dumping all the KV would be pretty crazy for stuff like `tokenizer.ggml.tokens` also `general.datasets` urls won't fit into one line... so there's that.

I think we could figure out a more suitable approach, but the main point of this is for a sanity check by developers that all the layers and kv is present as expected.

Reply:
I agree with truncating arrays (because the token list is very big), but not with truncating the strings within.

Why not truncate `general.description` to `This gguf is p...aykeye attempt` then? Feels to me like the same reason. (especially since `llama-cli` trims it to `This gguf is ported from a first vers...`)

I think the strings inside lists of strings should not be truncated in the markdown dump.